src/x509.rs

```diff
@@ -157,6 +157,74 @@ impl<S: BERDecodable + Integer, A: BERDecodable + SignatureAlgorithm, K: BERDeco
     }
 }
 
+/// BasicConstraints type as defined in RFC 5280.
+///
+/// It is recommended to use the `ca()` or `no_ca()` constructors instead of constructing
+/// instances of this type directly, to ensure the following statement from the RFC is enforced:
```

Comment:
Why can't we make the fields private?

Reply:
To give clients more freedom and to allow them to get the info form badly constructed certificates.

The alternative would indeed be to make the fields private and to return the literal value from the `path_len_constraint()` getter method. The only downside is that it is then completely up to the client to enforce the "The `pathLenConstraint` field is meaningful only if the `cA` boolean is asserted [...]" statement.